sites/bestbuy.py

```diff
@@ -6,327 +6,567 @@
     from Cryptodome.Cipher import PKCS1_OAEP
 from base64 import b64encode
 from utils import send_webhook, random_delay
-import requests, time, lxml.html, json, sys, settings
+from selenium import webdriver
+from webdriver_manager.chrome import ChromeDriverManager
+from selenium.webdriver.chrome.options import Options
+from selenium.common import exceptions
+import requests, time, lxml.html, json, sys, settings, urllib3, threading
+
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
 
 
-# TODO: Implement settings.dont_buy global as to not checkout
 class BestBuy:
-    def __init__(self,task_id,status_signal,image_signal,product,profile,proxy,monitor_delay,error_delay):
-        self.status_signal,self.image_signal,self.product,self.profile,self.monitor_delay,self.error_delay = status_signal,image_signal,product,profile,float(monitor_delay),float(error_delay)
+
+    def __init__(self, task_id, status_signal, image_signal, product, profile, proxy, monitor_delay, error_delay):
+        self.status_signal, self.image_signal, self.product, self.profile, self.monitor_delay, self.error_delay = status_signal, image_signal, product, profile, float(
+            monitor_delay), float(error_delay)
         self.session = requests.Session()
+        self.sku_id = self.product.split('=')[1]
+        self.cookies_from_browser = self.get_cookies_using_browser
+        self.kill_cookie_thread = False
+        cookie_thread = threading.Thread(
+            target=self.launch_cookie_thread,
+            args=(),
+        )
+        cookie_thread.name = "cookieThread"
+        cookie_thread.daemon = True
+        cookie_thread.start()
+
         if proxy != False:
             self.session.proxies.update(proxy)
-        self.status_signal.emit({"msg":"Starting","status":"normal"})
-        tas_data = self.get_tas_data()
-        product_image = self.monitor()
-        self.atc()
-        self.start_checkout()
-        self.submit_shipping()
-        self.submit_payment(tas_data)
+        starting_msg = "Starting"
+        if settings.dont_buy:
+            starting_msg = "Starting in dev mode - Phoenix Bot will not actually checkout (dont_buy = True)"
+        self.status_signal.emit({"msg": starting_msg, "status": "normal"})
+
         while True:
-            success,jwt = self.submit_order()
-            if not success and jwt != None:
-                transaction_id = self.handle_3dsecure(jwt)
-                self.submit_card(transaction_id)
-            else:
-                if success:
-                    send_webhook("OP","Bestbuy",self.profile["profile_name"],task_id,product_image)
+            self.status_signal.emit({"msg": "Checking Stock", "status": "normal"})
+            in_stock = False
```

Comment:
I thought about it and it would actually be better to initialize in_stock to a call to self.check_stock(). That way if the item is in stock we skip the while loop. In other words we will only enter the while loop if the item is out of stock which is better.

```suggestion
            in_stock = self.check_stock()
```